### PR TITLE
Story/tg 12

### PR DESCRIPTION
**Pull request by disist**
Integration with heroku

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/garage-it/SmartHouse-frontend/pull/21" target="_blank">ReviewNinja

File: .travis.yml [hunk 1]
```diff
@@ -2,11 +2,17 @@ sudo: false
...
\ No newline at end of file
+  - codecov
+deploy:
+  provider: heroku
+  api_key:
+    secure: YWk9sRKKIQxl6AeRvsbNhXGTHaa48U0+gRTdyrFYU2NUz2igGPZ+9faAqmwJBcgvwcSQhc0mIXSlc9rZ+4doOKl16kVYXihFNtG1m0dchN/xT/t7sPA1knVg/81tuRyiuQoNYqgyYT1fdJftXek1Rovt5Zk3D+CNE2UEM3B8ps2nIBAWX/f5nTFxWu7y2S4NP9MEslXhfDNQuXhW7pZIVDOQ5A2jGjc3Pt6BzqegUl4569Ytuf9MPqy0DN5LNVNGOqMaOx6C3KkvTN8GdRktEATMvFUwbeh6xUwUJiCHxp2nIRjh8rKTEjgcQA4f0ibzz4LbqE3vm+2QS19qFiKniybFmTifETc+6ABgAj62BDqaseiBb7YospN3e6RXbJ6oFdI8qiaz/oZvpv1WvVO9blkI7/p7k92aRUncb0bqynJR2Tftn1Jrdbz2XM61RQ8GKvHc3tJWDgE6osdF3lko3ZD3pUGpMbmHcSo8LMPFmCe+jAfuawLU3M3j6LnmkNsmxVZZ+JES1FH0aA7CQjeg1NY2l4q27Z3y71+a58+lvQCGsn1d/J5gAneNSaodpljirFKbaGYcHxkB1Q4UdARn18atAFpn5V5RpS5IIwmIuhpErrxJE/PXRR1dOmwCvjgzGNmMYc3lOpgIas3uN8cR9nFQtWZpo7S43fMkocBrGPI=
+  app: garage-it-smart-house
+  on:
+    repo: garage-it/SmartHouse-frontend
+    branch: story/TG-12
```
Comment (yurii-sorokin) [inline .travis.yml]: master?

File: test/index.js [hunk 2]
```diff
@@ -20,12 +20,17 @@ import 'zone.js/dist/zone';
...
+// Config
+import envConfig from '../env/env.config';
+
 testing.setBaseTestProviders(
     browser.TEST_BROWSER_PLATFORM_PROVIDERS,
     browser.TEST_BROWSER_APPLICATION_PROVIDERS);
 
 Object.assign(global, testing);
 
+window.ENV_PUBLIC_CONFIG = envConfig.public;
```
Comment (yurii-sorokin) [inline test/index.js]: global.ENV_PUBLIC_CONFIG?

File: .travis.yml (hunk 1 above)
Comment (disist) [inline .travis.yml]: You absolutely right. Thanks!

File: test/index.js (hunk 2 above)
Comment (disist) [inline test/index.js]: Could you please clarify.

PS: I need to change this on 'global.ENV_PUBLIC_CONFIG' ?

Comment (yurii-sorokin) [inline test/index.js]: Yeap. It's our node environment, so it's better to use global.

Comment (disist) [inline test/index.js]: Aaa, okay, no problem. Fixed

File: env/env.config.js [hunk 3]
```diff
@@ -1,36 +1,4 @@
-const path = require('path');
-const rootDir = path.resolve(__dirname, '..');
+const ENV_CONFIG = process.env.ENV_CONFIG || 'default';
+const envConfig = require(`./env.${ENV_CONFIG}.config`);
```
Comment (kosich) [inline env/env.config.js]: naming is a bit strange

File: .travis.yml [hunk 4]
```diff
@@ -2,11 +2,16 @@ sudo: false
...
   - npm run build
-
 after_success:
-  - codecov
\ No newline at end of file
+  - codecov
+deploy:
+  provider: heroku
+  api_key:
+    secure: YWk9sRKKIQxl6AeRvsbNhXGTHaa48U0+gRTdyrFYU2NUz2igGPZ+9faAqmwJBcgvwcSQhc0mIXSlc9rZ+4doOKl16kVYXihFNtG1m0dchN/xT/t7sPA1knVg/81tuRyiuQoNYqgyYT1fdJftXek1Rovt5Zk3D+CNE2UEM3B8ps2nIBAWX/f5nTFxWu7y2S4NP9MEslXhfDNQuXhW7pZIVDOQ5A2jGjc3Pt6BzqegUl4569Ytuf9MPqy0DN5LNVNGOqMaOx6C3KkvTN8GdRktEATMvFUwbeh6xUwUJiCHxp2nIRjh8rKTEjgcQA4f0ibzz4LbqE3vm+2QS19qFiKniybFmTifETc+6ABgAj62BDqaseiBb7YospN3e6RXbJ6oFdI8qiaz/oZvpv1WvVO9blkI7/p7k92aRUncb0bqynJR2Tftn1Jrdbz2XM61RQ8GKvHc3tJWDgE6osdF3lko3ZD3pUGpMbmHcSo8LMPFmCe+jAfuawLU3M3j6LnmkNsmxVZZ+JES1FH0aA7CQjeg1NY2l4q27Z3y71+a58+lvQCGsn1d/J5gAneNSaodpljirFKbaGYcHxkB1Q4UdARn18atAFpn5V5RpS5IIwmIuhpErrxJE/PXRR1dOmwCvjgzGNmMYc3lOpgIas3uN8cR9nFQtWZpo7S43fMkocBrGPI=
```
Comment (kosich) [inline .travis.yml:14]: can this be stored publicly?

File: env/env.config.js (hunk 3 above)
Comment (disist) [inline env/env.config.js]: Thanks, I will fix this in next commit

File: .travis.yml (hunk 4 above)
Comment (disist) [inline .travis.yml:14]: Yep, it's common solution when deploy to heroku from travis.

File: env/env.common.config.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+const rootDir = path.resolve(__dirname, '..');
+
+function rootPath(...args) {
+    return path.join(rootDir, ...args);
+}
+
+module.exports = {
+    path: rootPath,
+    staticServer: {
+        host: 'localhost',
```
Comment (artem713) [inline env/env.common.config.js:11]: Can you please move it to constant

File: env/env.common.config.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        port: 8000
+    },
+    mockServer: {
+        host: 'localhost',
```
Comment (artem713) [inline env/env.common.config.js:15]: Can you please move it to constant

File: env/env.default.config.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+const envConfig = require('./env.common.config');
+
+module.exports = Object.assign(envConfig, {
+    public: {
+        backEndUrl: '//localhost:3000',
+        backEndWebSocketUrl: '//localhost:3000'
```
Comment (artem713) [inline env/env.default.config.js:6]: Can you please move it to constant

File: env/env.heroku.config.js [hunk 8]
```diff
@@ -0,0 +1,8 @@
+const envConfig = require('./env.common.config');
+
+module.exports = Object.assign(envConfig, {
+    public: {
+        backEndUrl: '//garage-it-smart-house-api.herokuapp.com',
+        backEndWebSocketUrl: '//garage-it-smart-house-api.herokuapp.com'
```
Comment (artem713) [inline env/env.heroku.config.js:6]: Can you please move it to constant

File: env/env.config.js (hunk 3 above)
Comment (disist) [inline env/env.config.js]: Fixed

File: env/env.common.config.js (hunk 5 above)
Comment (disist) [inline env/env.common.config.js:11]: This is constants file. 

Could you please clarify, what you mean about constant?

File: env/env.common.config.js (hunk 6 above)
Comment (disist) [inline env/env.common.config.js:15]: please see comment above

File: env/env.default.config.js (hunk 7 above)
Comment (disist) [inline env/env.default.config.js:6]: please see comment above

File: env/env.heroku.config.js (hunk 8 above)
Comment (disist) [inline env/env.heroku.config.js:6]: please see comment above

**Pull request merged**